.github/workflows/main.yml

```
@@ -365,3 +365,44 @@ jobs:
             && echo "Untracked files: ${untracked:-<none>}"                     \
             && test -z "${untracked}"
         shell: bash
+  benchmark:
+    name: Run benchmark suite
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    needs: build
+    steps:
+      # Check out the code
+      - name: Download Artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: built-tree
+      - name: Extract Artifact
+        run: |-
+          echo "::group::Untar Archive"
+          tar zxvf built-tree.tgz
+          echo "::endgroup"
+
+          rm built-tree.tgz
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          cache: yarn
+          node-version: '16'
+      - name: Install Dependencies
+        run: |-
+          yarn install --frozen-lockfile
+      - name: Run Benchmark
+        run: |-
+          cd packages/@jsii/benchmarks
+          yarn bench | tee output.txt
+      - name: Store Benchmark Results
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: JSII Benchmark
+          tool: 'benchmarkjs'
+          output-file-path: packages/@jsii/benchmarks/output.txt
+          comment-always: true
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          fail-on-alert: true
+          auto-push: false
```

Instead of `cd`, you should use the `working-directory` property.

I'd also emit the results out-of-tree because it's cleaner.

```suggestion
      - name: Run Benchmark
        run: |-
          yarn bench | tee ${{ runner.temp }}/bench-output.txt
        working-directory: packages/@jsii/benchmarks
      - name: Store benchmark result
        uses: benchmark-action/github-action-benchmark@v1
        with:
          name: JSII Benchmark
          tool: 'benchmarkjs'
          output-file-path: ${{ runner.temp }}/bench-output.txt
          comment-always: true
          github-token: ${{ secrets.GITHUB_TOKEN }}
          fail-on-alert: true
          auto-push: false
```